pkg/client/client_test.go

```diff
@@ -59,133 +66,248 @@ func TestApplySingle(t *testing.T) {
+		t.Run("as corev1 objects", func(t *testing.T) {
+
+			t.Run("when using forceUpdate=true", func(t *testing.T) {
+
+				t.Run("it should not update when specs are same", func(t *testing.T) {
+					// given
+					cl, _ := newClient(t, s)
+					obj := defaultService.DeepCopy()
+					_, err := cl.CreateOrUpdateObject(obj, true, nil)
+					require.NoError(t, err)
+					originalGeneration := obj.GetGeneration()
+
+					// when updating with the same obj again
+					createdOrChanged, err := cl.CreateOrUpdateObject(obj, true, nil)
+
+					// then
+					require.NoError(t, err)
+					assert.False(t, createdOrChanged) // resource was not updated on the server, so returned value is `false`
+					updateGeneration := obj.GetGeneration()
+					assert.Equal(t, originalGeneration, updateGeneration)
+				})
+
+				t.Run("it should not update when specs are same except ClusterIP", func(t *testing.T) {
+					// given
+					cl, _ := newClient(t, s)
+					obj := defaultService.DeepCopy()
+					_, err := cl.CreateOrUpdateObject(obj, true, nil)
+					require.NoError(t, err)
+					originalGeneration := obj.GetGeneration()
+					obj.Spec.ClusterIP = "" // modify for version to update
+					// when updating with the same obj again
+					createdOrChanged, err := cl.CreateOrUpdateObject(obj, true, nil)
+
+					// then
+					require.NoError(t, err)
+					assert.False(t, createdOrChanged) // resource was not updated on the server, so returned value is `false`
+					updateGeneration := obj.GetGeneration()
+					assert.Equal(t, originalGeneration, updateGeneration)
+					assert.Equal(t, defaultService.Spec.ClusterIP, obj.Spec.ClusterIP)
+				})
+
+				t.Run("it should update when specs are different", func(t *testing.T) {
+					// given
+					cl, _ := newClient(t, s)
+					obj := defaultService.DeepCopy()
+					_, err := cl.CreateOrUpdateObject(obj, true, nil)
+					require.NoError(t, err)
+					originalGeneration := obj.GetGeneration()
+
+					// when updating with the modified obj
+					modifiedObj := modifiedService.DeepCopy()
+					modifiedObj.Spec.ClusterIP = ""
+					modifiedObj.ObjectMeta.Generation = obj.GetGeneration()
+					createdOrChanged, err := cl.CreateOrUpdateObject(modifiedObj, true, nil)
+
+					// then
+					require.NoError(t, err)
+					assert.True(t, createdOrChanged) // resource was updated on the server, so returned value if `true`
+					updateGeneration := modifiedObj.GetGeneration()
+					assert.Equal(t, originalGeneration+1, updateGeneration)
+				})
+
+				t.Run("it should update when specs are different including ClusterIP", func(t *testing.T) {
+					// given
+					cl, _ := newClient(t, s)
+					obj := defaultService.DeepCopy()
+					_, err := cl.CreateOrUpdateObject(obj, true, nil)
+					require.NoError(t, err)
+					originalGeneration := obj.GetGeneration()
+
+					// when updating with the modified obj
+					modifiedObj := modifiedService.DeepCopy()
+					modifiedObj.ObjectMeta.Generation = obj.GetGeneration()
```

Comment:
the same here - we don't need to copy that:
```suggestion
```

Reply:
done in 83e733b